distributed/security.py

```diff
@@ -153,18 +153,57 @@ def _set_field(self, kwargs, field, config_name):
             out = dask.config.get(config_name)
         setattr(self, field, out)
 
-    def __repr__(self):
+    def attr_to_dict(self):
         keys = sorted(self.__slots__)
         keys.remove("extra_conn_args")
-        items = []
+
+        attr = {}
+
         for k in keys:
             val = getattr(self, k)
             if val is not None:
                 if isinstance(val, str) and "\n" in val:
-                    items.append((k, "..."))
+                    attr[k] = "Temporary (In-memory)"
+                elif isinstance(val, str):
+                    attr[k] = f"Local ({os.path.abspath(val)})"
                 else:
-                    items.append((k, repr(val)))
-        return "Security(" + ", ".join(f"{k}={v}" for k, v in items) + ")"
+                    attr[k] = val
+
+        return attr
+
+    def __repr__(self):
+        attr = self.attr_to_dict()
+        return (
+            "Security("
+            + ", ".join(f"{key}={value}" for key, value in attr.items())
```

Comment:
```suggestion
            + ",\n    ".join(f"{key}={value}" for key, value in attr.items())
```

What do you think adding newlines to improve reading/scanning the text?

Reply:
Oh, I see above you experimented with using a dictionary to get a similar kind of behaviour. Yes, I like the way that looks a lot more - it's much easier to skim through the information and find what you need.